src/pubsub/ua_pubsub_readergroup.c

```diff
@@ -565,17 +591,32 @@ UA_Server_setReaderGroupOperational(UA_Server *server,
                 return retval;
             }
         }
+
+    }
+    UA_UNLOCK(&server->serviceMutex);
+    return ret;
+}
 #endif
-        ret = UA_ReaderGroup_setPubSubState(server, rg, UA_PUBSUBSTATE_OPERATIONAL,
+
+UA_StatusCode
+UA_Server_enableReaderGroup(UA_Server *server, const UA_NodeId readerGroupId){
+    UA_LOCK(&server->serviceMutex);
+    UA_StatusCode ret = UA_STATUSCODE_BADNOTFOUND;
+    UA_ReaderGroup *rg = UA_ReaderGroup_findRGbyId(server, readerGroupId);
+    if(rg)
+    {
+        // TODO: Check connection state to decide the taget state
```

Comment:
Resolve the TODO.

Reply:
This would be resolved in the pull request #5706